### PR TITLE
add dev container

### PR DESCRIPTION
**Pull request by ilaner**
## Status

- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues

fixes: https://github.com/demisto/etc/issues/39504

## Description
add dev container to extension

## Screenshots

Paste here any images that will help the reviewer

## Does it break backward compatibility?

- [ ] Yes
  - Further details:
- [ ] No

## Must have

- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by jochman — CHANGES_REQUESTED** [review]
Update Changelog
Update README

File: README.md [hunk 1]
```diff
@@ -39,6 +39,25 @@ The linters will write its data to the path configured in `xsoar.autoFindProblem
...
+
+There are two options:
+
+1. Run content repository on Dev Container. This opens the repository on [demisto-sdk docker](https://github.com/demisto/dockerfiles/tree/master/docker/demisto-sdk), which contains `demisto-sdk` and content `content` basic requirements.
+   To activate, run the command `Open content in Dev Container` from the command pallette or right click in file or editor.
+2. Run an integration of script on dev container. This opens a workspace inside a container that is based upon the integration or script docker image (which is specified in their YAML file). This workspace is fully configured with `Python`, `Pylance`, `flake8`, `mypy` and `pytest`, allowing developing and debugging inside the integration environment.
+   To activate, run the command `Open integration/script in Dev Container` from the command pallette or right click in file or editor inside a specific integration or a script.
+   If running an integration which is on `python 2`, the installation of `mypy` will fail because it's not supported on `python 2`. In addition, in order to debug the tests it is required to install `Python` extension version `2022.2`:
```
Comment (jochman) [inline README.md:52]: add a not that we are not planning to support python2 and it is just a workaround until it disabled.

File: README.md [hunk 2]
```diff
@@ -39,6 +39,25 @@ The linters will write its data to the path configured in `xsoar.autoFindProblem
...
+   ![image](https://user-images.githubusercontent.com/88267954/165749900-0a152f16-f715-4144-ab79-0d34f16b1a6a.png)
+   ![image](https://user-images.githubusercontent.com/88267954/165750018-897fb6ac-b6e2-48a5-ad4b-e7d4533bb8b2.png)
+
+* Make sure you have `ms-vscode-remote.remote-containers` extension installed.
```
Comment (jochman) [inline README.md:57]: add link

File: README.md [hunk 3]
```diff
@@ -39,6 +39,25 @@ The linters will write its data to the path configured in `xsoar.autoFindProblem
...
+* Make sure you have `docker daemon` running.
```
Comment (jochman) [inline README.md:58]: add a way to check it out. something like `docker ps`

File: README.md [hunk 4]
```diff
@@ -39,6 +39,25 @@ The linters will write its data to the path configured in `xsoar.autoFindProblem
...
+* It is possible to open content in Dev Container, and inside this container open integration/script in Dev Container.
+* The integration/script Dev Container does not have `demisto-sdk` or `git`.
+* The workspace folder is mirrored with the local folder. It is possible to work simultaneously on the same files locally and with Dev Container.
```
Comment (jochman) [inline README.md:61]: Use also a professional term (bind or attached via volume) along with "mirrored".

File: README.md [hunk 5]
```diff
@@ -39,6 +39,25 @@ The linters will write its data to the path configured in `xsoar.autoFindProblem
...
+   To activate, run the command `Open integration/script in Dev Container` from the command pallette or right click in file or editor inside a specific integration or a script.
+   If running an integration which is on `python 2`, the installation of `mypy` will fail because it's not supported on `python 2`. In addition, in order to debug the tests it is required to install `Python` extension version `2022.2`:
+   ![image](https://user-images.githubusercontent.com/88267954/165749900-0a152f16-f715-4144-ab79-0d34f16b1a6a.png)
+   ![image](https://user-images.githubusercontent.com/88267954/165750018-897fb6ac-b6e2-48a5-ad4b-e7d4533bb8b2.png)
+
+* Make sure you have `ms-vscode-remote.remote-containers` extension installed.
+* Make sure you have `docker daemon` running.
+* It is possible to open content in Dev Container, and inside this container open integration/script in Dev Container.
+* The integration/script Dev Container does not have `demisto-sdk` or `git`.
```
Comment (jochman) [inline README.md:60]: Write what is the problem with that.

File: Templates/devcontainer.json [hunk 6]
```diff
@@ -18,9 +18,7 @@
 			"--ignore-missing-imports",
 			"--show-column-numbers",
 			"--no-pretty",
-			"--allow-redefinition",
-			"--install-types",
-			"--non-interactive"
```
Comment (jochman) [inline Templates/devcontainer.json]: why not add the `--non-interactive` flag?

File: CHANGELOG.md [hunk 7]
```diff
@@ -10,7 +10,7 @@
 - Removed the support of the **maintenance** option from the *Update Release Notes* command.
 - Fixed running demisto-sdk commands in menus from directories to run in the correct directory.
 - Added an option to run lint with tests in menu.
-
+- Added an option to run content and integrations/scripts on development container.
```
Comment (jochman) [inline CHANGELOG.md:13]:
```suggestion
- Added an option to run content and integrations/scripts in a development container with right-click and pick <command-name>.  also add a gif/screenshot of it.
```

**Review by jochman — CHANGES_REQUESTED** [review]
Cool! 🍺

File: Templates/devcontainer.json (hunk 6 above)
Comment (ilaner) [reply Templates/devcontainer.json]: For some reason, when adding this flags the mypy problems are not shown in `problems`. The only reason to use `--non-interactive` is if we want to use `--install-types` which dynamically installs the mypy required types.

File: Templates/devcontainer.json [hunk 8]
```diff
@@ -3,7 +3,7 @@
 	"build": {
 		"dockerfile": "Dockerfile",
 		"args": {
-			"IMAGENAME": "demisto/demisto-sdk:1.6.1.27477",
+			"IMAGENAME": "demisto/demisto-sdk:latest",
```
Comment (jochman) [inline Templates/devcontainer.json:6]: But there is no latest...

**Review by jochman — APPROVED** [review]

**Pull request merged**